cpp/include/cuspatial/distance/point_linestring_distance.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_view.hpp>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute distance between pairs of points and linestrings (a.k.a. polylines)
+ *
+ * The distance between a point and a linestring is defined as the minimum distance
+ * between the point and any segment of the linestring. For each input point, this
+ * function returns the distance between the point and the corresponding linestring.
+ *
+ * The following example contains 2 pairs of points and linestrings.
+ * ```
+ * First pair:
+ * Point: (0, 0)
+ * Linestring: (0, 1) -> (1, 0) -> (2, 0)
+ *
+ * Second pair:
+ * Point: (1, 1)
+ * Linestring: (0, 0) -> (1, 1) -> (2, 0) -> (3, 0) -> (3, 1)
+ *
+ * The input of the abbove example is:
+ * points_x: {0, 1}
+ * points_y: {0, 1}
+ * linestring_offsets: {0, 3}
+ * linestring_x: {0, 1, 2, 0, 1, 2, 3, 3}
+ * linestring_y: {1, 0, 0, 0, 1, 0, 0, 1}
+ *
+ * Result: {sqrt(2)/2, 0}
+ * ```
+ *
+ * @param points_x x-coordinates of points.
+ * @param points_y y-coordinates of points.
+ * @param linestring_offsets Indices of the start of each linestring in the `linestring_x` and
+ * `linestring_y` arrays.
```

Comment:
In fact, this PR hasn't updated to adapting to the new geoarrow format yet. There was dilemma over whether I should refactor all PRs in one go or refactor every PR while we can. Since a request is made here. I'll work on that.

Reply:
I think that it works without modification, because the original algorithms only look for the beginning of the last offset group, whereas GeoArrow includes the ending of the last offset group, one additional value. I'm interested to find out in this if it works without modification.

Reply:
Two places needs modification: the number of pairs were taken from the size of the offsets arrays, this needs minus 1. The ternaries in some kernels can also be removed because we can guarantee no OB access now. Finally the checks should be changed.

Reply:
But all minor changes.